### ISSUE TITLE
Training constructors for LARS

### ISSUE THREAD

**Issue by micyril** [mlpack/mlpack#948]
Right now LARS has two constructors
```
  LARS(const bool useCholesky = false,
       const double lambda1 = 0.0,
       const double lambda2 = 0.0,
       const double tolerance = 1e-16);

  LARS(const bool useCholesky,
       const arma::mat& gramMatrix,
       const double lambda1 = 0.0,
       const double lambda2 = 0.0,
       const double tolerance = 1e-16);
```
and one training method
```
  void Train(const arma::mat& data,
             const arma::vec& responses,
             arma::vec& beta,
             const bool transposeData = true);
```
Every classification and regression algorithm I have encountered in mlpack has a training constructor (a constructor with a body containing a call of a Train method), what is considered to be a good practice according to the design guidelines. But there is a lack of such ones for LARS. Should we consider adding some training constructors to LARS? I can imagine such signatures:
```
  LARS(const arma::mat& data,
       const arma::vec& responses,
       const bool transposeData = true,
       const bool useCholesky = false,
       const double lambda1 = 0.0,
       const double lambda2 = 0.0,
       const double tolerance = 1e-16);

  LARS(const arma::mat& data,
       const arma::vec& responses,
       const bool transposeData,
       const bool useCholesky,
       const arma::mat& gramMatrix,
       const double lambda1 = 0.0,
       const double lambda2 = 0.0,
       const double tolerance = 1e-16);
```

**Comment by rcurtin** [comment]
I agree, this would be a welcome improvement.

**Comment by micyril** [comment]
Ok, I will handle it

**Comment by rcurtin** [comment]
Fixed with #957.
